src/data-access/repositories/mongo/mongo-entity-converter.ts

```diff
@@ -209,15 +280,25 @@ export class MongoEntityConverter {
   }
 
   public convertToMongoPart(part: Part): MongoPart {
+    let timings: PartTimings | undefined = undefined
+    try {
+      timings = part.getTimings()
+    } catch (error) {
+      if ((error as Exception).errorCode !== ErrorCode.UNSUPPORTED_OPERATION) {
+        throw error
+      }
+    }
```

Comment:
By extracting this to its own method, both methods would be simpler and you can avoid working with mutable variables.